client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java

```diff
@@ -0,0 +1,99 @@
+
+    private Profile bookProfile;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_book);
+        // Get database for profile
+        final DatabaseWrapper dbw = DatabaseWrapper.getWrapper();
+
+        // Set buttons and warning field to views
+        final Button cancelButton = findViewById(R.id.newBookCancelButton);
+        final Button saveButton = findViewById(R.id.newBookSaveButton);
+
+        // The fields of the book
+        final EditText bookTitle = findViewById(R.id.newBookTitle);
+        final EditText bookDesc = findViewById(R.id.newBookDesc);
+        final EditText bookAuthor = findViewById(R.id.newBookAuthor);
+        final EditText bookISBN = findViewById(R.id.newBookISBN);
+        // TODO: replace editTExt with an actual image
+        final EditText bookImageURL = findViewById(R.id.newBookImageURL);
+        final TextView warning = findViewById(R.id.newBookWarning);
+
+        // Get the current profile
+        dbw.getProfile(dbw.userId).addOnSuccessListener(new OnSuccessListener<Profile>() {
+            @Override
+            public void onSuccess(Profile profile) {
+                bookProfile = profile;
+            }
+        });
+
+        // Save button is clicked
+        saveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!bookTitle.getText().toString().isEmpty()
+                        && !bookDesc.getText().toString().isEmpty()
+                        && !bookAuthor.getText().toString().isEmpty()
+                        && !bookISBN.getText().toString().isEmpty()) {
+
+                    // Create the book
+                    Book newBook = new Book(bookProfile.getUserID(),
+                            bookTitle.getText().toString(),
+                            bookAuthor.getText().toString(),
+                            bookDesc.getText().toString(),
+                            bookISBN.getText().toString(),
+                            "Available");
+
+                    // Adds in image url if present
+                    if (!bookImageURL.getText().toString().isEmpty()) {
+                        newBook.setImageURL(bookImageURL.getText().toString());
+                    }
+
+                    // Adds new book to database
+                    dbw.addBook(newBook);
+                    // Go back to my books
+                    Snackbar.make(findViewById(R.id.popupMessage), R.string.popup_confirm, Snackbar.LENGTH_SHORT).show();
+                    bookTitle.setText("");
+                    bookAuthor.setText("");
+                    bookDesc.setText("");
+                    bookISBN.setText("");
+                } else {
+                    // Don't create a new book
+                    warning.setText(R.string.book_warning);
```

Comment:
after negating the condition and moving this block
then we could add a return statement after this
there would be no need to indent the other block